client/src/client.ts

```diff
@@ -161,7 +161,7 @@ function createConnection(input: any, output: any, errorHandler: ConnectionError
 		sendNotification: (type: string | RPCMessageType, params?: any): void => connection.sendNotification(Is.string(type) ? type : type.method, params),
 		onNotification: (type: string | RPCMessageType, handler: GenericNotificationHandler): void => connection.onNotification(Is.string(type) ? type : type.method, handler),
 
-		trace: (value: Trace, tracer: Tracer, sendNotification: boolean = false): void => connection.trace(value, tracer, sendNotification),
+		trace: (value: Trace, tracer: Tracer, sendNotification: boolean = false, traceFormat: TraceFormat = TraceFormat.Text): void => connection.trace(value, tracer, sendNotification, traceFormat),
```

Comment:
I know it is easier to implement if we simply add an optional arg at the endd but I would prefer to have to signatures like this:

```
trace: (value: Trace, tracer: Tracer, sendNotification: boolean = false)
trace: (value: Trace, traceFormat: TraceFormat, tracer: Tracer, sendNotification: boolean = false)
```

to keep the args together and may be deprecate the old signature.

Reply:
@dbaeumer Decided to have this:

```ts
trace(value: Trace, tracer: Tracer, sendNotification?: boolean): void;
trace(value: Trace, tracer: Tracer, traceOptions?: TraceOptions): void;

// In json-rpc

export interface TraceOptions {
	sendNotification?: boolean;
	traceFormat?: TraceFormat;
}
```

I initially had what you suggested, but later abandoned it in favor of `traceFormat?` at the end, as
- I'd prefer to keep `traceFormat` optional
- A call with three args can be of either signature.

With this new one it's easier to detect the params of the overloaded functions, and in the future it's easier to add more options without breaking backward compatibility.

Reply:
No, this is not what I meant. I was suggesting to have this:

```ts
trace: (value: Trace, tracer: Tracer, sendNotification: boolean = false)
trace: (value: Trace, traceFormat: TraceFormat, tracer: Tracer, sendNotification: boolean = false)
```

with one implementation overload like this:

```ts
trace(arg0: Trace, arg1: Tracer | TraceFormat, arg2?: Tracer | boolean, arg3?: boolean)
```

There is enough information in the signatures to decide which on is meant.